### PR TITLE
External editors: make gui visible in darkroom

### PR DESCRIPTION
**Pull request by Mark-64**
With the recent introduction of export module in darkroom it comes in handy to make ext_editor.lua GUI visible in darkroom as well.
With this, we can export to collection and make round trips to our favorite program, all without leaving darkroom and not being forved to rely on keyboard shortcut only.

### CODE REVIEW

File: contrib/ext_editor.lua [hunk 1]
```diff
@@ -409,27 +410,11 @@ local box1 = dt.new_widget("box") {
...
   }
 
+
+-- install module in lighttable and darkroom ----------------------------------
 table.insert(ee.widgets, combobox)
 table.insert(ee.widgets, box1)
-
--- register new module "external editors" in lighttable ------------------------
-if dt.gui.current_view().id == "lighttable" then
-  install_module()
```
Comment (wpferguson) [inline contrib/ext_editor.lua:417]: Have you tried starting darktable and having it open an image directly, I.e.darktable /path/to/an/image.xxx ?  The reason this is in there is that adding a module to lighttable, when you're not in llighttable view, crashes darktable.

Comment (Mark-64) [reply contrib/ext_editor.lua:417]: Indeed! 
I misunderstood the purpose of that part of code, sorry about that. 
Reverted now

**Comment by wpferguson** [comment]
Should displaying the lib be an option, or make it the default but give the user a preference they can set to not display in darkroom?  Just a thought, it's up to you.

**Comment by Mark-64** [comment]
good idea, will do

File: contrib/ext_editor.lua [hunk 2]
```diff
@@ -334,30 +332,50 @@ local function export2collection(storage, image_table, extra_data)
...
-      {[dt.gui.views.lighttable] = {"DT_UI_CONTAINER_PANEL_RIGHT_CENTER", 100}},  
-      dt.new_widget("box") {
-        orientation = "vertical",
-        table.unpack(ee.widgets),
-        },
-      nil,  -- view_enter
-      nil   -- view_leave
-      )
+    if dr then
+      -- register new module "external editors" in lighttable and darkroom ----
```
Comment (wpferguson) [inline contrib/ext_editor.lua:343]: you could do 

```
local views = {[dt.gui.views.lighttable] = {"DT_UI_CONTAINER_PANEL_RIGHT_CENTER", 100}}
if dr then 
views = {[dt.gui.views.lighttable] = {"DT_UI_CONTAINER_PANEL_RIGHT_CENTER", 100},
         [dt.gui.views.darkroom] = {"DT_UI_CONTAINER_PANEL_LEFT_CENTER", 100}}
end
```

then just replace the view part with the variable views and so only have the one code block for installing the module

Comment (Mark-64) [reply contrib/ext_editor.lua]: implemented

**Comment by wpferguson** [comment]
Thanks
\

**Pull request merged**